MobileBuy/sample/src/main/java/com/shopify/sample/activity/DiscountActivity.java

```diff
@@ -112,13 +111,8 @@ public void onClick(View v) {
 
         OptimizelyClient optimizelyClient = app.getOptimizelyManager().getOptimizely();
         String userId = app.getUser();
-        String cartCTAExperimentKey = getIntent().getStringExtra("optlyCartCtaExperimentKey");
-        String userAttributesString = getIntent().getStringExtra("optlyUserAttributes");
-        HashMap<String, String> userAttributes = new HashMap<>();
-        if (userAttributesString != null) {
-            Gson gson = new Gson();
-            userAttributes = (HashMap<String, String>) gson.fromJson(userAttributesString, userAttributes.getClass());
-        }
+        String cartCTAExperimentKey = app.getOptimizelyCTAExperimentKey();
```

Comment:
the purpose of the intent system in Android, is that each intent can be an entry point with the context necessary to progress through the app. This means that users could be deep linked to screens in an Android app with information such as `optlyCartCtaExpeirmentKey` passed to that screen of the app so that information would be available locally.

The way this *could* play out in our e2e app is that we could jump directly to the checkout screen to see what the CTA would be there without having to click through the app. We could have the appium driver launch the app at just this activity and not have to click through multiple screens to get here. 

This is a long explanation of another method we could try for the android app. Instead of having the initial launch screen be a singleton for all information in the app, we could have appium launch the app in this screen and just check the CTA here in this one activity.